CMakeLists.txt

```diff
@@ -102,15 +115,38 @@ set(TARGET_NAME slim)
 file(GLOB_RECURSE SLIM_SOURCES ${PROJECT_SOURCE_DIR}/core/*.cpp ${PROJECT_SOURCE_DIR}/eidos/*.cpp)
 add_executable(${TARGET_NAME} ${SLIM_SOURCES})
 target_include_directories(${TARGET_NAME} PRIVATE ${GSL_INCLUDES} "${PROJECT_SOURCE_DIR}/core" "${PROJECT_SOURCE_DIR}/eidos")
-target_link_libraries(${TARGET_NAME} PUBLIC gsl)
-target_link_libraries(${TARGET_NAME} PUBLIC tables)
-target_link_libraries(${TARGET_NAME} PUBLIC)
+target_link_libraries(${TARGET_NAME} PUBLIC gsl tables)
 
 set(TARGET_NAME eidos)
 file(GLOB_RECURSE EIDOS_SOURCES  ${PROJECT_SOURCE_DIR}/eidos/*.cpp  ${PROJECT_SOURCE_DIR}/eidostool/*.cpp)
 add_executable(${TARGET_NAME} ${EIDOS_SOURCES})
 target_include_directories(${TARGET_NAME} PRIVATE ${GSL_INCLUDES} "${PROJECT_SOURCE_DIR}/eidos")
-target_link_libraries(${TARGET_NAME} PUBLIC gsl)
-target_link_libraries(${TARGET_NAME} PUBLIC tables)
+target_link_libraries(${TARGET_NAME} PUBLIC gsl tables)
+
+set(TARGET_NAME eidoslib)
+file(GLOB_RECURSE EIDOS_SOURCES  ${PROJECT_SOURCE_DIR}/eidos/*.cpp  ${PROJECT_SOURCE_DIR}/eidostool/*.cpp)
+add_library(${TARGET_NAME} ${EIDOS_SOURCES})
+target_include_directories(${TARGET_NAME} PRIVATE ${GSL_INCLUDES} "${PROJECT_SOURCE_DIR}/eidos")
 
 install(TARGETS slim eidos DESTINATION bin)
+
+# QtSLiM -- this can be enabled with the -DBUILD_QTSLIM=ON option to cmake
+if(BUILD_QTSLIM)
+set(TARGET_NAME qtslim)
+find_package(Qt5 REQUIRED
+    Core
+    Gui
+    Widgets
+)
+set(CMAKE_AUTOMOC ON)
+set(CMAKE_AUTORCC ON)
+set(CMAKE_AUTOUIC ON)
+list(REMOVE_ITEM SLIM_SOURCES ${PROJECT_SOURCE_DIR}/core/main.cpp)
```

Comment:
Ah, I see you're removing the slim tool's main() here, good.  The whole project structure is probably a bit wonky really.  Maybe the issue I mention above with the eidos tool's main() is also handled already somehow?

Reply:
The eidostool objects are not linked into qtlslim at all, should they be?

Reply:
Probably not; all of the lower-level sources need to be compiled with the QtSLiM-specific compile flags turned on, so the objects from the other builds can't really be re-used.